File: src/test/resources/vertx-js/buffer.js
var io = Packages.io;
var JBuffer = io.vertx.core.buffer.Buffer;
var Buffer = function(j_val) {
    var j_buffer = j_val;
    var that = this;
    this.appendString = function(str) {
        j_buffer.appendString(str);
        return that;
    };
    this.toString = function(enc) {
        return j_buffer.toString(enc);
    };
};
Buffer.buffer = function(s) {
    return new Buffer(JBuffer.buffer(s));
};
module.exports = Buffer;

File: src/test/resources/vertx/buffer.rb
module Vertx
  class Buffer
    def initialize(j_del)
      @j_del = j_del
    end
    def self.buffer(s)
      ::Vertx::Buffer.new((Java::IoVertxCoreBuffer::Buffer.buffer(s)))
    end
    def append_string(str)
      @j_del.appendString(str)
    end
    def to_string(enc)
      @j_del.toString(enc)
    end
  end
end
